proseco/guide_report.py

```diff
@@ -0,0 +1,290 @@
+    vmax = np.ceil(vmax / 100) * 100
+    ax.imshow(region.transpose(), cmap='hot', origin='lower', vmin=1, vmax=vmax,
+              extent=(row0, row0 + hw * 2, col0, col0 + hw * 2))
+
+    # Plot a box showing the 8x8 image boundaries
+    x = rowc - 4
+    y = colc - 4
+    patch = patches.Rectangle((x, y), 8, 8, edgecolor='y', facecolor='none', lw=1)
+    ax.add_patch(patch)
+
+    # Make an empty box (no spoilers) or a cute 8x8 grid
+    if len(spoilers) == 0:
+        plt.text(rowc, colc, "No Spoilers", color='y', fontweight='bold',
+                 ha='center', va='center')
+    else:
+        for i in range(1, 8):
+            ax.plot([x, x + 8], [y + i, y + i], color='y', lw=1)
+            ax.plot([x + i, x + i], [y, y + 8], color='y', lw=1)
+
+    # Add mag text to each spoiler
+    for spoil in spoilers:
+        # Mag label above or below to ensure the label is in the image
+        dcol = -3 if (spoil['col'] > colc) else 3
+        plt.text(spoil['row'], spoil['col'] + dcol, f"mag={spoil['mag']:.1f}",
+                 color='y', fontweight='bold', ha='center', va='center')
+
+    plt.title(f"Spoiler stars (vmax={vmax:.0f})")
+    ax.set_xlabel('Row')
+    ax.set_ylabel('Column')
+
+
+def plot_imposters(guide, cand_guides, ax):
+    """
+    Make the hot pixel imposters plot for a given `guide` candidate.
+    """
+    # Figure out pixel region for dithered-over-pixels plot
+    row_extent = np.ceil(4 + cand_guides.dither.row)
+    col_extent = np.ceil(4 + cand_guides.dither.col)
+    rminus, rplus = get_ax_range(guide['row'], row_extent)
+    cminus, cplus = get_ax_range(guide['col'], col_extent)
+
+    # Pixel region of the guide
+    img = ACAImage(np.zeros(shape=(rplus - rminus, cplus - cminus)),
+                   row0=rminus, col0=cminus)
+    dark = ACAImage(cand_guides.dark, row0=-512, col0=-512)
+    img += dark.aca
+
+    # Pixel region of the guide plus some space for annotation
+    row0 = rminus - 4
+    col0 = cminus - 4
+    drow = (rplus - rminus) + 8
+    dcol = (cplus - cminus) + 8
+    canvas = ACAImage(np.zeros(shape=(drow, dcol)), row0=row0, col0=col0)
+    canvas += img.aca
+
+    ax.imshow(canvas.transpose(), interpolation='none', cmap='hot', origin='lower',
+              vmin=50, vmax=3000, extent=(row0, row0 + drow, col0, col0 + dcol))
+
+    # If force excluded, will not have imposter mag
+    if not (guide['forced'] and guide['stage'] == -1):
+        # Add max region with "mag"
+        x = guide['imp_r']
+        y = guide['imp_c']
+        patch = patches.Rectangle((x, y), 2, 2, edgecolor='y', facecolor='none', lw=1.5)
+        ax.add_patch(patch)
+        plt.text(row0 + drow / 2, col0 + dcol - 1, f"box 'mag' {guide['imp_mag']:.1f}",
+                 ha='center', va='center', color='y', fontweight='bold')
+
+    # Plot a box showing the 8x8 image boundaries
+    x = row0 + drow / 2 - 4
+    y = col0 + dcol / 2 - 4
+    patch = patches.Rectangle((x, y), 8, 8, edgecolor='y', facecolor='none', lw=1)
```

Comment:
I was also trying to decide if there was value in trying to plot some indicator of the edge of the useable fov, though that's probably just applicable for the force-included stars that concerned me earlier.